modules/devices/dmi_memory.c

```diff
@@ -382,6 +385,34 @@ static void dmi_fill_from_spd(dmi_mem_socket *s) {
         s->type_detail = g_strdup(s->spd->type_detail);
 }
 
+static dmi_mem_size size_of_online_memory_blocks() {
+    gchar **online_files = NULL;
+    gchar *block_size_bytes_str = NULL;
+    dmi_mem_size block_size_bytes = 0;
+    dmi_mem_size ret = 0;
+
+    if (g_file_get_contents("/sys/devices/system/memory/block_size_bytes", &block_size_bytes_str, NULL, NULL) ) {
+        block_size_bytes = strtoll(block_size_bytes_str, NULL, 16);
+    }
+    if (!block_size_bytes)
+        return 0;
+
+    online_files = get_output_lines("find /sys/devices/system/memory -name online");
```

Comment:
I would prefer not to call any shell command unless it's absolutely necessary.

Reply:
Ok